engines/stark/ui/menu/settingsmenu.cpp

```diff
@@ -0,0 +1,405 @@
+	_widgets.back()->setupSounds(3, 4);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"HighResHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+	
+	_widgets.push_back(new CheckboxWidget(
+			"Subtitles",
+			StarkSettings->getBoolSetting(Settings::kSubtitle),
+			CLICK_HANDLER(SettingsMenuScreen, flipSettingHandler<Settings::kSubtitle>),
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kSubtitles>)));
+	_widgets.back()->setupSounds(3, 4);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"SubtitleHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new CheckboxWidget(
+			"SpecialFX",
+			StarkSettings->getBoolSetting(Settings::kSpecialFX),
+			CLICK_HANDLER(SettingsMenuScreen, flipSettingHandler<Settings::kSpecialFX>),
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kSpecialFX>)));
+	_widgets.back()->setupSounds(3, 4);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"SpecialFXHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new CheckboxWidget(
+			"Shadows",
+			StarkSettings->getBoolSetting(Settings::kShadow),
+			CLICK_HANDLER(SettingsMenuScreen, flipSettingHandler<Settings::kShadow>),
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kShadows>)));
+	_widgets.back()->setupSounds(3, 4);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"ShadowsHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new CheckboxWidget(
+			"HighResFMV",
+			StarkSettings->getBoolSetting(Settings::kHighFMV),
+			CLICK_HANDLER(SettingsMenuScreen, flipSettingHandler<Settings::kHighFMV>),
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kHighResFMV>)));
+	_widgets.back()->setupSounds(3, 4);
+	_widgets.back()->setVisible(StarkSettings->hasLowResFMV());
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"FMVHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new StaticLocationWidget(
+			"VSettings",
+			nullptr,
+			nullptr));
+
+	_widgets.push_back(new VolumeWidget(
+			"Voice",
+			_cursor,
+			_soundManager, 0,
+			Settings::kVoice,
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kVoice>)));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"VoiceHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new VolumeWidget(
+			"Music",
+			_cursor,
+			_soundManager, 2,
+			Settings::kMusic,
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kMusic>)));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"MusicHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	_widgets.push_back(new VolumeWidget(
+			"Sfx",
+			_cursor,
+			_soundManager, 1,
+			Settings::kSfx,
+			MOVE_HANDLER(SettingsMenuScreen, textHandler<kSfx>)));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"SfxHelp",
+			nullptr,
+			nullptr));
+	_widgets.back()->setVisible(false);
+
+	if (!isDemo()) {
```

Comment:
This test should be different. My 4 CD French version does not have `AllowFF` either. Perhaps test if the render entry exists?